src/transaction/components/TransactionProvider.tsx

```diff
@@ -69,26 +69,54 @@ export function TransactionProvider({
     }
   }, [contracts, writeContract]);
 
+  const switchChain = useCallback(
```

Comment:
To keep this code easier to mantain and test, we should probably split those into indipendent custom Hooks.

Let me know how much this is douable or not.